app/apollo/test/channel.fga.spec.js

```diff
@@ -0,0 +1,666 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { expect } = require('chai');
+const fs = require('fs');
+const { v4: UUID } = require('uuid');
+const { MongoMemoryServer } = require('mongodb-memory-server');
+
+const { models } = require('../models');
+const resourceFunc = require('./api');
+const channelFunc = require('./channelApi');
+
+const apollo = require('../index');
+const { AUTH_MODEL } = require('../models/const');
+
+// If the current auth model does not support FGA, skip FGA unit tests without error.
+if (AUTH_MODEL === 'extauthtest' || AUTH_MODEL === 'passport.local') {
+  console.log(`Found non fine-grained auth model: ${AUTH_MODEL}. Skipping fine-grained auth tests.`);
+  return process.exit(0);
+}
+const externalAuth = require('../../externalAuth.js');
+const testHelperPath = externalAuth.ExternalAuthModels[AUTH_MODEL] ? externalAuth.ExternalAuthModels[AUTH_MODEL].testPath : `./testHelper.${AUTH_MODEL}`;
+const { prepareUser, prepareOrganization, signInUser } = require(testHelperPath);
+const testDataPath = externalAuth.ExternalAuthModels[AUTH_MODEL] ? externalAuth.ExternalAuthModels[AUTH_MODEL].testDataPath : `./app/apollo/test/data/${AUTH_MODEL}`;
+
+let mongoServer;
+let myApollo;
+
+const graphqlPort = 18000;
+const graphqlUrl = `http://localhost:${graphqlPort}/graphql`;
+const resourceApi = resourceFunc(graphqlUrl);
+const channelApi = channelFunc(graphqlUrl);
+
+let fgaToken01;
+let fgaToken02;
+
+let org01Data;
+let org01;
+
+let fgaUser01Data;
+let fgaUser02Data;
+
+const createOrganizations = async () => {
+  org01Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/cluster.spec.org_01.json`,
+      'utf8',
+    ),
+  );
+  org01 = await prepareOrganization(models, org01Data);
+};
+
+const createUsers = async () => {
+  fgaUser01Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/fga.spec.user01.json`,
+      'utf8',
+    ),
+  );
+  await prepareUser(models, fgaUser01Data);
+  fgaUser02Data = JSON.parse(
+    fs.readFileSync(
+      `${testDataPath}/fga.spec.user02.json`,
+      'utf8',
+    ),
+  );
+  await prepareUser(models, fgaUser02Data);
+  return {};
+};
+
+const createChannels = async () => {
+  await models.Channel.create({
+    _id: 'fake_ch_id_1',
+    org_id: org01._id,
+    uuid: 'testConfiguration1',
+    name: 'test-configuration1',
+    versions: [],  /* channel versions is deprecated and no longer used */
+    tags: ['tag_01'],
+    contentType: 'local'
+  });
+  await models.Channel.create({
+    _id: 'fake_ch_id_2',
+    org_id: org01._id,
+    uuid: 'testConfiguration2',
+    name: 'test-configuration2',
+    versions: [],  /* channel versions is deprecated and no longer used */
+    tags: ['tag_02'],
+    contentType: 'local'
+  });
+};
```

Comment:
The 'out of the box' channels (and versions and groups later) should be defined as constants so that they can be referenced later, e.g.:
```
const testChannel1 = {
    _id: UUID(),   // generate a UUID -- tests should not be able to assume to know the database-internal _id value
    org_id: org01._id,
    uuid: 'testChannel1-uuid',    // use a value that makes it clear that it's a UUID not a name
    name: 'testChannel1-name',    // use a value that makes it clear that it's a name not a UUID
    versions: [],  /* channel versions is deprecated and no longer used */
    tags: ['tag_01'],
    contentType: 'local'
  });
```
(the `//` comments after some lines do not need to be kept -- there just for explanation in this example)

then
```
const createChannels = async () => {
  await models.Channel.create( testChannel1 );
  [...]
```

Then later, when tests need to use channel 1, instead of things like
```
      } = await channelApi.channelByName(fgaToken01, {
        orgId: org01._id,
        name: 'test-configuration1',
      });
```
the code can reuse the constants rather than having to duplicate (and potentially misspell or mis-capitalize or whatever):
```
      } = await channelApi.channelByName(fgaToken01, {
        orgId: org01._id,
        name: testChannel1.name,
      });
```
This is why `org01` is used as a constant as well, it's a better code pattern rather than duplicating string literals in multiple places.

Reply:
This is done for fga channel unit tests. Working on others

Reply:
Resolved (for channel fga tests at least, other pending)

Reply:
Done for all fga unit tests

Reply:
Resolved